homeassistant/components/tuya/climate.py

```diff
@@ -430,9 +430,10 @@ def fan_mode(self) -> str | None:
     @property
     def fan_modes(self) -> list[str]:
         """Return fan modes for select."""
-        return json.loads(
-            self.tuya_device.function.get(DPCODE_FAN_SPEED_ENUM, {}).values
-        ).get("range")
+        fan_speed_device_function = self.tuya_device.function.get(DPCODE_FAN_SPEED_ENUM)
+        if not fan_speed_device_function:
+            return []
+        return json.loads(fan_speed_device_function.values).get("range")
```

Comment:
Thanks! I'm not sure what the return value type is here. If this is also a `dict.get` call the type annotation isn't correct. Also just a side note, that can be fixed separately.

Reply:
The function get returns a TuyaDevice instance

![image](https://user-images.githubusercontent.com/195327/135993109-28cd41c1-c8d0-481c-8676-213e17363a39.png)

The values returns a dict
![image](https://user-images.githubusercontent.com/195327/135993399-329164bc-c5eb-452a-8c8e-22d2c9487434.png)

I'm not sure if I see your concern

Reply:
I mean, after we load the the values with `json.loads`. If the return value of that is a dict, `dict.get` may return `None` and our type annotation wouldn't be complete.

Reply:
Side note: `values` can't be a dict, since then `json.loads` wouldn't work.

Reply:
Alright, ensured the default return is an empty list, to make our return type correct.